clang/lib/Sema/SemaBounds.cpp

```diff
@@ -4069,8 +4134,10 @@ namespace {
         const VarDecl *W = Pair.first;
         BoundsExpr *Bounds = Pair.second;
         BoundsExpr *AdjustedBounds = ReplaceVariableInBounds(Bounds, V, OriginalValue, CSS);
-        if (!Pair.second->isUnknown() && AdjustedBounds->isUnknown())
+        if (!Pair.second->isUnknown() && AdjustedBounds->isUnknown()) {
```

Comment:
This seems to only update BlameAssignments if the updated observed bounds of W are unknown. For a case like:

```
unsigned int i;
array_ptr<int> p : count(i) = 0;
// Original value of i in i++ is i - 1
// i++ causes the updated observed bounds of p to be (p, p + i - 1)
i++, unrelated_variable = 0;
```

We want `BlameAssignments[p] => i++`.